### PR TITLE
Get PROXY protocol guide to match the style guide

### LINKED ISSUE #32168: Edit the PROXY protocol guide to match the Teleport style guide
## Applies To

`docs/pages/management/security/proxy-protocol.mdx`

## Details

We merged this guide without a round of copy edits. Let's make sure it abides by the docs [style guide](https://goteleport.com/docs/contributing/documentation/style-guide).

## How will we know this is resolved?

The PROXY protocol guide abides by the Teleport style guide.

### PR DESCRIPTION
**Pull request by ptgott**
Closes #32168

Format the guide as a how-to guide (rather than a reference or architectural guide). This is appropriate, since the guide is a detailed look at a single configuration option that requires some architectural planning for users can enable it.

Also add an introductory section that describes the scope of the guide, and make various edits for readability and consistency with the rest of the docs site.

### CODE REVIEW

File: docs/pages/management/security/proxy-protocol.mdx [hunk 1]
```diff
@@ -1,88 +1,134 @@
 ---
-title: PROXY protocol
-description: How to securely configure PROXY protocol usage with Teleport
+title: Run Teleport Behind a Layer 4 Load Balancer
```
Comment (zmb3) [inline docs/pages/management/security/proxy-protocol.mdx:2]: Feels like the title should mention proxy protocol, no?

File: docs/pages/management/security/proxy-protocol.mdx [hunk 2]
```diff
@@ -1,88 +1,134 @@
...
+description: Describes how to securely configure Teleport to use the PROXY protocol so you can run the Auth Service and Proxy Service behind a layer 4 load balancer.
```
Comment (zmb3) [inline docs/pages/management/security/proxy-protocol.mdx:3]: Don't we just use PROXY protocol to preserve the original client IP? If so, it's not necessarily required to run behind a L4 LB.

File: docs/pages/management/security/proxy-protocol.mdx [hunk 3]
```diff
@@ -1,88 +1,134 @@
...
+reliable client IP information is important from a security standpoint,
+because features like audit logging and IP pinning depend on it. If the PROXY
+protocol is not configured correctly, these features will be compromised.
 
-An example of the PROXYv1 protocol header:
+<Notice type="note">
+
+Users of Teleport Enterprise (Cloud) do not need to manage PROXY protocol
+setting. Teleport-managed Auth Service and Proxy Service deployments run behind
+an L4 load balancer with the PROXY protocol configured.
```
Comment (zmb3) [inline docs/pages/management/security/proxy-protocol.mdx:19]: @sclevine is this technically accurate?

Comment (sclevine) [reply docs/pages/management/security/proxy-protocol.mdx:19]: We have two layers of load balancing:
- AWS NLBs, which are non-terminating L4 load balancers
- Envoy, which is a connection (but not TLS) terminating load balancer (harder to categorize in OSI)

We do use PROXY protocol.

File: docs/pages/management/security/proxy-protocol.mdx (hunk 1 above)
Comment (sclevine) [reply docs/pages/management/security/proxy-protocol.mdx:2]: Agree that it makes more sense for this guide to focus on proxy protocol and not L4 load balancing in general. There are other ways to achieve source IP preservation with L4 load balancers (e.g., AWS NLBs are L4 load balancers that can do source IP preservation with SNAT).

File: docs/pages/management/security/proxy-protocol.mdx (hunk 3 above)
Comment (zmb3) [reply docs/pages/management/security/proxy-protocol.mdx:19]: Thanks for confirming :+1:

**Review by zmb3 — APPROVED** [review]

**Review by stevenGravy — APPROVED** [review]

**Pull request merged**